File: 3133-minimum-array-end/3133-minimum-array-end.java
class Solution {
    public long minEnd(int n, int x) {

        long num = x , one = 1;

        List<Integer> unsetBits = new ArrayList<>();
        for(int bit = 0 ; bit < 63 ; bit++){
            if((num >> bit & one ^ one) == 1){
                unsetBits.add(bit);
            }
        }

        long mask = n - 1;
        for(int index = 0 ; index < unsetBits.size() ; index++){
            if((mask >> index & one) == 1){
                num |= one << unsetBits.get(index);
            }
        }
        return num;
    }
}

File: 3133-minimum-array-end/3133-minimum-array-end.js
/**
 * @param {number} n
 * @param {number} x
 * @return {number}
 */
var minEnd = function(n, x) {
    let num = BigInt(x) , one = BigInt(1);

    let unsetBits = [];
    for (let bit = 0n ; bit < 63n ; bit++) {
        if(num >> bit & one ^ one) {
            unsetBits.push(bit);
        }
    }

    let mask = BigInt(n - 1);
    for (let index = 0n ; index < unsetBits.length ; index++) {
        if (mask >> index & one) {
            console.log(unsetBits[index]);
            num |= one << unsetBits[index];
        }
    }
    return Number(num);
};